docs/reference/mapping/types/wildcard.asciidoc

```diff
@@ -0,0 +1,52 @@
+[role="xpack"]
+[testenv="basic"]
+[[wildcard]]
+=== Wildcard datatype
+++++
+<titleabbrev>Wildcard</titleabbrev>
+++++
+
+A `wildcard` field stores values optimised for wildcard queries.
+Wildcard queries are possible on other field types but suffer from constraints:
+* `text` fields limit matching of any wildcard expressions to individual tokens rather than the original whole value held in a field
+* `keyword` fields are untokenized but slow at performing wildcard queries (especially patterns with leading wildcards).
+
+Internally the `wildcard` field indexes the whole field value using ngrams and stores the full string in compressed chunks.
+The index is used as a rough filter to cut down the number of values that are then checked by retrieving and checking the full values from the compressed store.
+Storage costs are typically lower than those of `keyword` fields
```

Comment:
Maybe instead mention that this field is well suited for log lines, where it is both more efficient for grep-like queries and more space-efficient.

I worry that making the statement too generic will be a wrong incentive that will encourage some users to use this field instead of `keyword` when it's the wrong trade-off.